File: testers/avro/java-spring/generated-and-checked-in/com/example/events/AddressListener.java
package com.example.events;

import java.util.concurrent.CompletableFuture;
import org.apache.kafka.clients.consumer.ConsumerRecord;
import org.apache.kafka.common.header.Headers;
import org.springframework.kafka.annotation.KafkaListener;

/** Event listener interface for address topic. Implement this interface to handle events. */
public interface AddressListener {
  /** Receive and dispatch events to handler methods */
  @KafkaListener(topics = "address")
  default CompletableFuture<Void> receive(ConsumerRecord<String, Object> record) {
    return switch (record.value()) {
      case null -> onUnknown(record);
      case Address e -> onAddress(e, record.headers());
      default -> onUnknown(record);
    };
  }

  /** Handle Address event */
  CompletableFuture<Void> onAddress(Address event, Headers metadata);

  /** Handle unknown event types. Override to customize behavior. */
  default CompletableFuture<Void> onUnknown(ConsumerRecord<String, Object> record) {
    return CompletableFuture.completedFuture(null);
  }
}

File: testers/avro/scala-cats/generated-and-checked-in/com/example/events/AddressListener.scala
package com.example.events

import cats.effect.IO
import fs2.kafka.ConsumerRecord
import fs2.kafka.Headers
import scala.annotation.unused

/** Event listener interface for address topic. Implement this interface to handle events. */
trait AddressListener {
  /** Receive and dispatch events to handler methods */
  @unused
  def receive(record: ConsumerRecord[String, Any]): IO[Unit] = {
    return record.value match {
      case null => onUnknown(record)
      case e: Address => onAddress(e, record.headers)
    }
  }

  /** Handle Address event */
  def onAddress(
    event: Address,
    metadata: Headers
  ): IO[Unit]

  /** Handle unknown event types. Override to customize behavior. */
  def onUnknown(record: ConsumerRecord[String, Any]): IO[Unit] = {
    return IO.unit
  }
}